### ISSUE TITLE
bug(node): a combination of @types/node 17.0.14 and TS 3.7.3 raises an error

### ISSUE THREAD

**Issue by antongolub** [DefinitelyTyped/DefinitelyTyped#58562]
A combination of @types/node and typescript: 3.7.3 yields:

```
../../node_modules/.registry.npmjs.org/@types/node/12.20.43/node_modules/@types/node/url.d.ts:137:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'URLSearchParams' must be of type '{ new (init?: string | string[][] | URLSearchParams | Record<string, string>): URLSearchParams; prototype: URLSearchParams; }', but here has type '{ new (init?: string | string[][] | URLSearchParams | Record<string, string>): URLSearchParams; prototype: URLSearchParams; toString(): string; }'.

137         var URLSearchParams: {
                ~~~~~~~~~~~~~~~

  ../../node_modules/.registry.npmjs.org/typescript/3.7.3/node_modules/typescript/lib/lib.dom.d.ts:16107:13
    16107 declare var URLSearchParams: {
                      ~~~~~~~~~~~~~~~
    'URLSearchParams' was also declared here.
```

Is there a good workaround for this now?

_Originally posted by @aaronatbissell in https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34960#issuecomment-1027198564_

relates #34960

**Comment by antongolub** [comment]
> @aaronatbissell,
> 
> I'm afraid that only TS 3.8.2+ seems to be working. This is due to `URLSearchParams` var declarations mismatch:
> 
> [v3.7.3 lib/dom.generated.d.ts#L16087](https://github.com/microsoft/TypeScript/blob/v3.7.3/src/lib/dom.generated.d.ts#L16087)
> 
> ```ts
> declare var URLSearchParams: {
>     prototype: URLSearchParams;
>     new(init?: string[][] | Record<string, string> | string | URLSearchParams): URLSearchParams;
> };
> ```
> 
> [v3.8.2 lib/dom.generated.d.ts#L16136](https://github.com/microsoft/TypeScript/blob/v3.8.2/src/lib/dom.generated.d.ts#L16136)
> 
> ```ts
> declare var URLSearchParams: {
>     prototype: URLSearchParams;
>     new(init?: string[][] | Record<string, string> | string | URLSearchParams): URLSearchParams;
>     toString(): string;
> };
> ```
> 
> Anyway, I'll try to fix it.
> 
> **UPD**
> 
> 1. it's technically possible
> 2. the fix literally looks like a piece of ***: we have to detect TS 3.8.x `lib.dom.d.ts` version by its side-effects to declare `URLSeachParams` var type:
> 
> ```ts
>         var URLSearchParams: typeof globalThis extends { SpeechRecognitionError: infer SpeechRecognitionError }  ?
>             // For TS <= 3.8.2 compatibility
>             // https://github.com/microsoft/TypeScript/compare/v3.7.3...v3.8.2#diff-dc0eab937d15e62545da3ed7b4f40ad6b24f15dd88fbc6ceda2bfb4ed8356eb0R15085
>             {
>                 prototype: URLSearchParams;
>                 new(init?: string[][] | Record<string, string> | string | URLSearchParams): URLSearchParams;
>             }: {
>                 prototype: URLSearchParams;
>                 new(init?: string[][] | Record<string, string> | string | URLSearchParams): URLSearchParams;
>                 toString(): string;
>             };
> ```

**Comment by paranthamanwp** [comment]
Same issue to me as well

../../node_modules/.registry.npmjs.org/@types/node/12.20.43/node_modules/@types/node/url.d.ts:137:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'URLSearchParams' must be of type '{ new (init?: string | string[][] | URLSearchParams | Record<string, string>): URLSearchParams; prototype: URLSearchParams; }', but here has type '{ new (init?: string | string[][] | URLSearchParams | Record<string, string>): URLSearchParams; prototype: URLSearchParams; toString(): string; }'.

137         var URLSearchParams: {
                ~~~~~~~~~~~~~~~

  ../../node_modules/.registry.npmjs.org/typescript/3.7.3/node_modules/typescript/lib/lib.dom.d.ts:16107:13
    16107 declare var URLSearchParams: {
                      ~~~~~~~~~~~~~~~
    'URLSearchParams' was also declared here.

**Comment by sandersn** [comment]
Does this fail with TS 3.8 or above? `@types` packages only support TS 3.8 and above right now, and older versions are expected to fix their dependencies to previously-published versions.

Also, 3.8 will no longer be supported when TS 4.6 releases in 3 weeks or so.

**Comment by antongolub** [comment]
In TS range [3.8.2, 4.6] everything works fine. The issue affects 3.7 users (and below).

**Comment by miherlosev** [comment]
https://github.com/DevExpress/testcafe-hammerhead/issues/2732

**Comment by martin-badin** [comment]
> i could hack it like this:

I think the better way how to fix it is to update TS version to 3.8 or above

**Comment by antongolub** [comment]
@aaronatbissell, @martin-badin, @paranthamanwp,

Сould any of you check @types/node v17.0.18+ with TS 3.7?

**Comment by antongolub** [comment]
@aaronatbissell

**Comment by aaronatbissell** [comment]
Yes - looks like it's working with TS 3.7.3!

**Comment by antongolub** [comment]
Ok, so I'm closing this. Fixed in #58619
